rpcs3/main.cpp

```diff
@@ -524,6 +528,33 @@ int main(int argc, char** argv)
 		Emu.SetConfigOverride(config_override_path);
 	}
 
+	std::string firmware_path;
+
+	if (parser.isSet(arg_installfw))
+	{
+		firmware_path = parser.value(installfw_option).toStdString();
+		if (!fs::is_file(firmware_path))
+		{
+			report_fatal_error(fmt::format("No firmware file found: %s", firmware_path));
+			return 1;
+		}
+	}
+
+	// Force install firmware first if specified through command-line
+	if (fs::is_file(firmware_path))
+	{
+		if (auto gui_app = qobject_cast<gui_application*>(app.data()))
+		{
+			main_window* main_window = gui_app->m_main_window;
+			if (!main_window)
+			{
+				report_fatal_error("Cannot install firmware, exiting !");
+				return 1;
+			}
+			main_window->InstallFirmware(QString::fromStdString(firmware_path));
```

Comment:
I don't really like this approach...
It would be better if you just passed the firmware path to the application and then let the application handle the installation.
The gui application would pass it to the main window which handles the installation.
The headless application should just install the firmware without all the gui mumbo jumbo used in main_window.cpp

Reply:
Ok, I will rework on this scheme and push a commit when ready.